### PR TITLE
Merge shared mem to 0.18.x

### PR DESCRIPTION
**Pull request by bbondy**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: js/stores/appStore.js [hunk 1]
```diff
@@ -238,14 +238,17 @@ const createWindow = (action) => {
     mainWindow.webContents.on('did-finish-load', (e) => {
       lastEmittedState = appState
       mainWindow.webContents.setZoomLevel(zoomLevel[toolbarUserInterfaceScale] || 0.0)
-      e.sender.send(messages.INITIALIZE_WINDOW,
-        {
+
+      const mem = muon.shared_memory.create({
```
Comment (bridiver) [inline js/stores/appStore.js]: isn't there already a commit for this?

**Comment by bsclifton** [comment]
@bbondy this should be for the 4.4.x branches, right? (since we already pulled this into 0.18.x)

File: js/stores/appStore.js (hunk 1 above)
Comment (bbondy) [inline js/stores/appStore.js]: Not in 0.18.x, this PR is against 0.18.x, so it wouldn't show this diff if there was.

**Comment by bsclifton** [comment]
Misunderstanding on my part 😄  I don't see the changes in master yet though, so we may still want to cherry-pick this commit and PR against master

**Comment by cezaraugusto** [comment]
I'm unable to save state after this change, getting this err:
```
Error saving app state:  TypeError: Cannot read property 'filter' of undefined
    at payload.perWindowState.forEach (/Users/cezaraugusto/dev/browser-laptop/app/sessionStore.js:72:46)
    at Array.forEach (<anonymous>)
    at Promise (/Users/cezaraugusto/dev/browser-laptop/app/sessionStore.js:71:30)
    at Promise (<anonymous>)
    at Object.module.exports.saveAppState (/Users/cezaraugusto/dev/browser-laptop/app/sessionStore.js:65:10)
    at saveAppState (/Users/cezaraugusto/dev/browser-laptop/app/sessionStoreShutdown.js:88:23)
    at Timeout.setTimeout (/Users/cezaraugusto/dev/browser-laptop/app/sessionStoreShutdown.js:154:9)
    at ontimeout (timers.js:386:14)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```

opened #10332

**Comment by bsclifton** [comment]
Should be fixed after Muon version is upgraded 👍  (all branches now updated- sorry I missed that)

**Comment by cezaraugusto** [comment]
confirmed working thanks

**Pull request merged**